### PR TITLE
WIP: work with packaged llvm

### PR DESCRIPTION
**Pull request by arnaudgelas**
if LLVM_ROOT is provided (which is the case with python scripts), it
should use the compiled llvm; else it looks for llvm in standard
location.

### CODE REVIEW

**Comment by arnaudgelas** [comment]
To be able to package oclint on linux, it should be possible to work with packaged version of LLVM.
With these modifications, I have managed to build oclint-core with llvm-3.7 on ubuntu 15.10, and it should also work on debian distribution. I could make try creating one debian docker instance to validate.

Note:
* we should validate that this change did not break the current build system
* check that the found llvm (when from package) respect a provided required version
* apply these changes for all modules

Any help, test, comment, would be highly appreciated!!

Thanks

**Comment by arnaudgelas** [comment]
The first attempt did not work... I have fixed and tested it locally...

**Comment by arnaudgelas** [comment]
@ryuichis I don't really get why tests do not compile with this PR... Any idea?

File: oclint-core/cmake/OCLintConfig.cmake [hunk 1]
```diff
@@ -40,19 +40,23 @@ SET(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/bin)
...
+else()
+	find_package(LLVM REQUIRED CONFIG)
+endif()
 
 INCLUDE_DIRECTORIES( ${LLVM_INCLUDE_DIRS} )
 LINK_DIRECTORIES( ${LLVM_LIBRARY_DIRS} )
 ADD_DEFINITIONS( ${LLVM_DEFINITIONS} )
 
+message(STATUS "Found LLVM ${LLVM_PACKAGE_VERSION}")
+message(STATUS "Using LLVMConfig.cmake in: ${LLVM_DIR}")
```
Comment (ryuichis) [inline oclint-core/cmake/OCLintConfig.cmake]: Don't see `LLVM_DIR` is set from CI log.

**Comment by ryuichis** [comment]
@arnaudgelas frankly, not quite sure why the tests failed on Travis CI. Maybe take another look tomorrow with a refreshed mindset.

**Pull request merged**